htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -134,7 +134,43 @@ class ProductFournisseur extends Product
 
 	public $packaging;
 
-
+	public $fields=array(
```

Comment:
The properties of class ProductFournisseur should be same than properties of class Product + new ones (the class extedns Product).
So i suggest to not declare the ->fields here (it is already declared into product) and just add into the constructor, the new fileds

$this->fields = array_merge($this->fields, array of new fields)

So the ProductFournisseur will avoid duplicate declaration and ProductFournisseur will always be upt to date compared to Product.

Reply:
Ok that what I did but I had to overwrite the getFieldList method as this method reflects the database fields.